logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/ClientRepository.kt

```diff
@@ -77,13 +80,19 @@ class ClientDataSource(
     override suspend fun registerMLSClient(clientId: ClientId, publicKey: ByteArray): Either<CoreFailure, Unit> =
         clientRemoteRepository.registerMLSClient(clientId, publicKey.encodeBase64())
 
-    override suspend fun saveNewClients(userId: UserId, clients: List<ClientId>): Either<CoreFailure, Unit> =
+    override suspend fun saveNewClients(userId: UserId, clients: List<ClientId>, deviceType: List<DeviceType>?): Either<CoreFailure, Unit> =
         userMapper.toUserIdPersistence(userId).let { userEntity ->
-            clients.map { ClientEntity(userEntity, it.value) }.let { clientEntityList ->
+            clients.map { ClientEntity(userEntity, it.value, deviceType?.get(clients.indexOf(it))?.name) }.let { clientEntityList ->
                 wrapStorageRequest { clientDAO.insertClients(clientEntityList) }
             }
         }
 
     override suspend fun registerToken(body: PushTokenBody): Either<NetworkFailure, Unit> = clientRemoteRepository.registerToken(body)
     override suspend fun deregisterToken(token: String): Either<NetworkFailure, Unit> = clientRemoteRepository.deregisterToken(token)
+    override suspend fun getClientsByUserId(userId: UserId): Either<StorageFailure, List<OtherUserClients>> =
+        wrapStorageRequest {
+            clientDAO.getClientsOfUserByQualifiedID(QualifiedIDEntity(userId.value, userId.domain))
```

Comment:
```suggestion
            clientDAO.getClientsOfUserByQualifiedID(MapperProvider().idMapper.toDaoModel(userId))
```

Here we should use the `idMapper` instance to do it (please don't take the suggestion as compilable) :D